File: frontend/src/types/decision.ts
export type RecommendationStatus = 'pending_review' | 'accepted' | 'rejected' | 'superseded'
export type RecommendationPriority = 'low' | 'medium' | 'high' | 'urgent'

export interface TaskDraft {
  title: string
  greenhouse_id: number
  crop_batch_id: number | null
  crop_type: string
  crop_rule_version: string
  priority: RecommendationPriority
  action_type: string
  description: string
  suggested_execute_before: string | null
  source_warning_id: number
  source_recommendation_id: number
  requires_manual_confirmation: true
}

export interface DecisionRecommendation {
  id: number
  warning_event_id: number
  greenhouse_id: number
  crop_batch_id: number | null
  crop_type: string
  crop_rule_version: string
  recommendation_code: string
  priority: RecommendationPriority
  action_type: string
  title: string
  rationale: string
  action_steps_json: string[]
  execute_before: string | null
  safety_note: string
  status: RecommendationStatus
  rule_version: string
  generated_at: string
  reviewed_at: string | null
  review_note: string | null
  task_draft: TaskDraft | null
}

export interface CurrentDecision {
  greenhouse_id: number
  crop_batch_id: number | null
  crop_type: string
  crop_rule_version: string
  crop_match_status: string
  cross_crop_warning: boolean
  cross_region_warning: boolean
  validation_status: string
  decision_scope: string
  forecast_strategy: string
  growth_stage: string | null
  data_source: string | null
  calibration_status: string
  current_risk_level: string | null
  future_risk_timeline: Array<Record<string, unknown>>
  recommendations: DecisionRecommendation[]
  disclaimer: string
}

File: backend/app/schemas/decision.py
from __future__ import annotations

from datetime import datetime
from typing import Literal

from pydantic import BaseModel, ConfigDict, Field


RecommendationStatus = Literal["pending_review", "accepted", "rejected", "superseded"]
Priority = Literal["low", "medium", "high", "urgent"]


class RecommendationReviewRequest(BaseModel):
    operator: str = Field(min_length=1, max_length=60)
    note: str | None = Field(default=None, max_length=500)


class RecommendationRejectRequest(RecommendationReviewRequest):
    note: str = Field(min_length=1, max_length=500)


class TaskDraft(BaseModel):
    title: str
    greenhouse_id: int
    crop_batch_id: int | None
    crop_type: str
    crop_rule_version: str
    priority: Priority
    action_type: str
    description: str
    suggested_execute_before: datetime | None
    source_warning_id: int
    source_recommendation_id: int
    requires_manual_confirmation: bool = True


class DecisionRecommendationData(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    warning_event_id: int
    greenhouse_id: int
    crop_batch_id: int | None
    crop_type: str
    crop_rule_version: str
    recommendation_code: str
    priority: Priority
    action_type: str
    title: str
    rationale: str
    action_steps_json: list[str]
    execute_before: datetime | None
    safety_note: str
    status: RecommendationStatus
    rule_version: str
    generated_at: datetime
    reviewed_at: datetime | None
    review_note: str | None
    task_draft: TaskDraft | None = None


class CurrentDecisionData(BaseModel):
    greenhouse_id: int
    crop_batch_id: int | None
    crop_type: str
    growth_stage: str | None
    data_source: str | None
    calibration_status: str
    crop_rule_version: str = "rules_unavailable"
    crop_match_status: str = "mismatched"
    cross_crop_warning: bool = True
    cross_region_warning: bool = True
    validation_status: str = "validated_on_public_dataset"
    local_calibration_status: str = "not_calibrated"
    decision_scope: str = "decision_support"
    forecast_strategy: str = "per_target_per_horizon_hybrid"
    current_risk_level: str | None
    future_risk_timeline: list[dict[str, object]] = Field(default_factory=list)
    recommendations: list[DecisionRecommendationData] = Field(default_factory=list)
    disclaimer: str = "本建议由规则系统根据环境监测和预测结果生成，仅供辅助决策，不代表设备已执行。"
